modules/base/tool.go

```diff
@@ -206,11 +206,26 @@ func HashEmail(email string) string {
 
 // AvatarLink returns avatar link by given email.
 func AvatarLink(email string) string {
-	if setting.DisableGravatar || setting.OfflineMode {
-		return setting.AppSubUrl + "/img/avatar_default.png"
+	var url string
+
+	if !setting.OfflineMode {
+		if setting.EnableFederatedAvatar && setting.LibravatarService != nil {
```

Comment:
I think `OfflineMode` should only be checked in `setting.go`, so when it is `true`, `setting.EnableFederatedAvatar` should be force to be false at first place. Then continue avatar option check in this `AvatarLink` function.